core/src/main/codegen/templates/Parser.jj

```diff
@@ -3955,7 +3959,32 @@ SqlCreate SqlCreate() :
     }
 }
 </#if>
-
+<#if parser.execStatementParserMethods?size != 0>
+/**
+ * Parses an EXEC statement
+ */
+SqlNode SqlExec() :
+{
+    final Span s;
+	SqlNode macro;
+	SqlNode source;
+}
+{
+	/** Macro has the same domain as Table so TableRef() works */
+	<EXEC> macro = TableRef() { s = span(); }	
+	(
+<#-- additional literal parser methods are included here -->
+<#list parser.execStatementParserMethods as method>
+		source = ${method}(s, macro)
+	|
+</#list>
+        source = OrderedQueryOrExpr(ExprContext.ACCEPT_QUERY)
+	)
+	{
+		return source;
```

Comment:
Can we move this logic into the execStatementParserMethod itself rather than outside? Note that methods defined in the parserImpls.ftl file still have access to all the functions in Parser.jj.

Reply:
In this case should we even have the source = ... line if we don't have any default logic?

Reply:
Since the whole code block is wrapped in an <if>,  by definition there should be a method to call so I feel like the default source setting is unnecessary as it shouldn't ever happen anyways.